code/scripting/api/objs/model.cpp

```diff
@@ -277,16 +277,16 @@ ADE_FUNC(__len, l_ModelTextures, NULL, "Number of textures on model", "number",
 ADE_INDEXER(l_ModelTextures, "texture", "number Index/string TextureName", "texture", "Model textures, or invalid modeltextures handle if model handle is invalid")
 {
 	modeltextures_h *mth = NULL;
-	int new_tex = -1;
+	texture_h* new_tex   = nullptr;
```

Comment:
@asarium Could you check this one, `new_tex` is optional so it looks that this could trigger access of a nullptr [below]?(https://github.com/scp-fs2open/fs2open.github.com/pull/1881/files#diff-1869a2e128315108bfa3546b77c94486R322)

Reply:
The usage shouldn't be reachable since it will only be called if that parameter is set but it's better to check that to avoid any problems.

I'll go through your comments and I'll submit a pull requests with the fixes.

Reply:
I submitted the changes here: #1938.